app/uk/gov/hmrc/apiplatformoutboundsoap/models/MessageRequest.scala

```diff
@@ -34,7 +34,7 @@ case class Addressing(from: String, to: String, replyTo: String, faultTo: String
   validate(from.trim != "", "addressing.from being empty")
 }
 
-case class PrivateHeader(name: String, value: Option[String]) {
+case class PrivateHeader(name: String, value: String) {
   validate(name.trim.length < 1024, "privateHeaders name is longer than 1024 characters")
-  validate(!(value.isDefined && value.get.trim.length > 1024), "privateHeaders value is longer than 1024 characters")
+  validate(value.trim.length < 1024, "privateHeaders value is longer than 1024 characters")
```

Comment:
I think this will reject a value of 1024 length and return a consequently misleading error message. I'd probably have unit tests for values of length 1023, 1024 and 1025 since 'off-by-one' errors are so common

Reply:
I've made sure my test covers 1025 chars and assert the test accordingly for error msg